zkevm-circuits/src/keccak_circuit/keccak_packed.rs

```diff
@@ -0,0 +1,1770 @@
+        );
+
+        KeccakPackedConfig {
+            q_enable,
+            q_first,
+            q_round,
+            q_absorb,
+            q_round_end,
+            is_final,
+            hash_rlc,
+            state,
+            cell_values,
+            absorb_from,
+            absorb_data,
+            absorb_result,
+            squeeze_packed,
+            round_cst,
+            normalize_3,
+            normalize_4,
+            normalize_6,
+            chi_base_table,
+            chi_ext_table,
+            pack_unpack_table,
+            _marker: PhantomData,
+        }
+    }
+
+    pub(crate) fn assign(
+        &self,
+        mut layouter: impl Layouter<F>,
+        _size: usize,
+        witness: &[KeccakRow<F>],
+    ) -> Result<(), Error> {
+        layouter.assign_region(
+            || "assign keccak rows",
+            |mut region| {
+                for (offset, keccak_row) in witness.iter().enumerate() {
+                    self.set_row(
+                        &mut region,
+                        offset,
+                        keccak_row.is_final,
+                        keccak_row.hash_rlc,
+                        keccak_row.state,
+                        keccak_row.absorb_data.clone(),
+                        keccak_row.squeeze_data.clone(),
+                        keccak_row.cell_values.clone(),
+                    )?;
+                }
+                Ok(())
+            },
+        )
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn set_row(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        is_final: u64,
+        hash_rlc: F,
+        state: [F; KECCAK_WIDTH],
+        absorb_data: AbsorbData,
+        squeeze_data: SqueezeData,
+        cell_values: Vec<F>,
+    ) -> Result<(), Error> {
+        let round = (offset + 24) % 25;
+
+        self.q_enable.enable(region, offset)?;
+
+        // q_first
+        if offset == 0 {
+            self.q_first.enable(region, offset)?;
+        }
+
+        // q_round
+        region.assign_fixed(
+            || format!("assign q_round {}", offset),
+            self.q_round,
+            offset,
+            || Ok(F::from(round != 24)),
+        )?;
+        // q_absorb
+        region.assign_fixed(
+            || format!("assign q_absorb {}", offset),
+            self.q_absorb,
+            offset,
+            || Ok(F::from(round == 24)),
+        )?;
+        // q_round_end
+        if offset > 0 && round == 24 {
+            self.q_round_end.enable(region, offset)?;
+        }
+        // is_final
+        region.assign_advice(
+            || format!("assign is_final {}", offset),
+            self.is_final,
+            offset,
+            || Ok(F::from(is_final)),
+        )?;
+        // hash_rlc
+        region.assign_advice(
+            || format!("assign hash_rlc {}", offset),
+            self.hash_rlc,
+            offset,
+            || Ok(hash_rlc),
+        )?;
```

Comment:
This will not even be needed in the future. You can (if you want) just use a `Expression::Constant(F::one())` and forget about passing the rlc and also comparing it to anything.

Also, the naming is not correct. This is the permutation hash. But definitely not the hash one. As we can have multiple calls to `keccak_f` per hash.

Reply:
I guess so, but it's only really used when in the last block for a certain hash so the name kind of makes sense no? Using the same name here as it would be called in the external interface for the keccak lookups would make sense (though don't know for sure what it is/will be).